test/external/dataplane/gn_connectivity.go

```diff
@@ -0,0 +1,395 @@
+			BeforeEach(func() {
+				toEndpointType = tcp.GlobalServiceIP
+				networking = framework.PodNetworking
+				egressIPType = subFramework.PodSelector
+			})
+
+			// Access from a pod on NonGatewayNodes to external apps is not supported for an external-app-connected cluster
+
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
+		})
+
+		// Access from a HostNetworking pod to external apps is not supported
+
+		When("a pod connects via TCP to the globalIP of a remote headless service", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.GlobalPodIP
+				networking = framework.PodNetworking
+				egressIPType = subFramework.ClusterSelector
+			})
+
+			// Access from a pod on NonGatewayNodes to external apps is not supported for an external-app-connected cluster
+
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
+		})
+
+	})
+
+	When("In a non-external-app-connected cluster", func() {
+		BeforeEach(func() {
+			cluster = getNonExternalClusterIndex(framework.TestContext.ClusterIDs)
+		})
+
+		When("a pod connects via TCP to the globalIP of a remote service", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.GlobalServiceIP
+				networking = framework.PodNetworking
+				egressIPType = subFramework.ClusterSelector
+			})
+
+			When("the pod is not on a gateway", func() {
+				verifyInteraction(framework.NonGatewayNode)
+			})
+
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
+		})
+
+		When("a pod matching an egress IP namespace selector connects via TCP to the globalIP of a remote service", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.GlobalServiceIP
+				networking = framework.PodNetworking
+				egressIPType = subFramework.NameSpaceSelector
+			})
+
+			When("the pod is not on a gateway", func() {
+				verifyInteraction(framework.NonGatewayNode)
+			})
+
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
+		})
+
+		When("a pod matching an egress IP pod selector connects via TCP to the globalIP of a remote service", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.GlobalServiceIP
+				networking = framework.PodNetworking
+				egressIPType = subFramework.PodSelector
+			})
+
+			When("the pod is not on a gateway", func() {
+				verifyInteraction(framework.NonGatewayNode)
+			})
+
+			When("the pod is on a gateway", func() {
+				verifyInteraction(framework.GatewayNode)
+			})
+		})
+
+		When("a pod with HostNetworking connects via TCP to the globalIP of a remote service", func() {
+			BeforeEach(func() {
+				toEndpointType = tcp.GlobalServiceIP
+				networking = framework.HostNetworking
+				egressIPType = subFramework.ClusterSelector
+			})
+
+			// Access from a pod with HostNetworking fails if a pod is not on NonGateway nodes
```

Comment:
Can you explain what is the problem with that use-case?

Reply:
It just fails to connect and I'm still not sure the reason. If we have the use case for these configurations, we need to investigate more and consider improvement.  I would like to make it just documented and leave as is.